File: launch_school/Programming_Foundations/small_problems/multiply_lists.rb
# def multiply_list(arr1, arr2)
#   [arr1, arr2].transpose.map {|a| a[0] * a[1]}.flatten
# end

def multiply_list(arr1, arr2)
  [arr1, arr2].transpose.map {|nested_array| nested_array.reduce(:*)}
end

p multiply_list([3, 5, 7], [9, 10, 11]) == [27, 50, 77]

File: launch_school/JS210-Fundamentals-of-Javascript/small_problems/easy4/multiply_lists.js
'use strict';

function multiplyList(arr1, arr2) {
  let multipliedList = [];

  for (let index = 0; index < arr1.length; index += 1) {
    multipliedList.push(arr1[index] * arr2[index]);
  }

  console.log(multipliedList);
  return multipliedList;
}

multiplyList([3, 5, 7], [9, 10, 11]);    // [27, 50, 77]
